### PR TITLE
Added device statuses and DeviceIsUp

### PR DESCRIPTION
**Pull request by scruzin**

### CODE REVIEW

File: model/device.go [hunk 1]
```diff
@@ -388,23 +402,21 @@ func IsMacAddress(mac string) bool {
...
-func (dev *Device) InputListWithID() []string {
-	ins := strings.ReplaceAll(dev.Inputs, " ", "")
-	split := strings.Split(ins, ",")
-	var out []string
-	for _, s := range split {
-		out = append(out, dev.Name+"."+s)
+// DeviceIsUp returns true if the device exists and is up, or false
+// otherwise. Up status is determined by checking the uptime variable
+// associated with the device. The device is considered to be up if
+// the uptime is less than twice the monitor period,
```
Comment (saxon-milton) [inline model/device.go:408]: Full stop

**Review by saxon-milton — APPROVED** [review]

File: model/device.go [hunk 2]
```diff
@@ -388,23 +402,21 @@ func IsMacAddress(mac string) bool {
...
+func DeviceIsUp(ctx context.Context, store datastore.Store, mac int64) (bool, error) {
+	dev, err := GetDevice(ctx, store, mac)
+	if err != nil {
+		return false, fmt.Errorf("could not get device: %w", err)
 	}
-	return out
-}
-
-func (dev *Device) OutputListWithID() []string {
```
Comment (saxon-milton) [inline model/device.go:402]: Ditto

File: model/device.go [hunk 3]
```diff
@@ -388,23 +402,21 @@ func IsMacAddress(mac string) bool {
 	return true
 }
 
-// Retain these for the sensor/actuator migration then remove.
-func (dev *Device) InputListWithID() []string {
```
Comment (saxon-milton) [inline model/device.go:392]: Was it intended that this be removed ?

**Review by saxon-milton — CHANGES_REQUESTED** [review]

File: model/device.go (hunk 3 above)
Comment (scruzin) [reply model/device.go:392]: Yes. It's dead code.

**Pull request closed without merging**